From 40eb4e8d70278962b484b6baea66efcf11e08f0c Mon Sep 17 00:00:00 2001
From: Alejandro Ochoa <alejandro.ochoa@intel.com>
Date: Tue, 29 Mar 2016 02:22:45 -0600
Subject: [PATCH] ASoc: rt5645: Reduce micbias output voltage

Micbias output voltage has been reduced from 2.7V to 1.8V.
The noise generated by the built-in microphone is reduced.
Micbias DAPM route must be configured in machine driver
instead of codec driver.

Change-Id: I74b189ed0254975b1a349da0f914866dc3fc220d
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-22149
Signed-off-by: Alejandro Ochoa <alejandro.ochoa@intel.com>
Reviewed-on: https://android.intel.com:443/487262
---
 sound/soc/codecs/rt5645.c                  | 9 ++-------
 sound/soc/intel/board/cht_cr_dpcm_rt5645.c | 6 ++++--
 2 files changed, 6 insertions(+), 9 deletions(-)

diff --git a/sound/soc/codecs/rt5645.c b/sound/soc/codecs/rt5645.c
index 4806e0a..0b39a82 100644
--- a/sound/soc/codecs/rt5645.c
+++ b/sound/soc/codecs/rt5645.c
@@ -37,8 +37,6 @@
 
 #include "rt5645.h"
 
-#define RT5645_DET_EXT_MIC 0
-/* #define USE_INT_CLK */
 #define JD1_FUNC
 /* #define ALC_DRC_FUNC */
 #define USE_ASRC
@@ -74,7 +72,7 @@ static struct rt5645_init_reg init_list[] = {
 
 #ifdef JD1_FUNC
 	{ RT5645_IRQ_CTRL2	, 0x0200 },
-	{ RT5645_MICBIAS	, 0x0008 },
+	{ RT5645_MICBIAS	, 0xC008 },
 	{ RT5645_GEN_CTRL3	, 0x1180 },
 	{ RT5645_CJ_CTRL3	, 0x4000 },
 #endif
@@ -1652,8 +1650,6 @@ static int rt5645_bst2_event(struct snd_soc_dapm_widget *w,
 	case SND_SOC_DAPM_POST_PMU:
 		snd_soc_update_bits(codec, RT5645_PWR_ANLG2,
 			RT5645_PWR_BST2_P, RT5645_PWR_BST2_P);
-		snd_soc_update_bits(codec, RT5645_PWR_ANLG2,
-			RT5645_PWR_MB1, RT5645_PWR_MB1);
 		break;
 
 	case SND_SOC_DAPM_PRE_PMD:
@@ -2166,7 +2162,6 @@ static const struct snd_soc_dapm_route rt5645_dapm_routes[] = {
 	{ "BST1", NULL, "Mic Det Power" },
 	{ "BST2", NULL, "IN2P" },
 	{ "BST2", NULL, "IN2N" },
-	{ "BST2", NULL, "micbias1" },
 
 	{ "INL VOL", NULL, "IN2P" },
 	{ "INR VOL", NULL, "IN2N" },
@@ -3060,7 +3055,7 @@ static int rt5645_probe(struct snd_soc_codec *codec)
 	snd_soc_update_bits(codec, RT5645_PWR_ANLG1, RT5645_LDO_SEL_MASK, 0x0);
 
 	/* dc_calibrate(codec); */
-	codec->dapm.bias_level = SND_SOC_BIAS_OFF;/*SND_SOC_BIAS_STANDBY;*/
+	codec->dapm.bias_level = SND_SOC_BIAS_OFF;
 	rt5645->codec = codec;
 	rt5645->combo_jack_en = true; /* enable combo jack */
 
diff --git a/sound/soc/intel/board/cht_cr_dpcm_rt5645.c b/sound/soc/intel/board/cht_cr_dpcm_rt5645.c
index e8095e7..c2ef383 100644
--- a/sound/soc/intel/board/cht_cr_dpcm_rt5645.c
+++ b/sound/soc/intel/board/cht_cr_dpcm_rt5645.c
@@ -486,12 +486,14 @@ static const struct snd_soc_dapm_route cht_audio_map[] = {
 	{"IN1P", NULL, "Headset Mic"},
 	{"IN1N", NULL, "Headset Mic"},
 	{"micbias1", NULL, "Int Mic"},
-	{"DMIC L1", NULL, "Int Mic"},
-	{"DMIC R1", NULL, "Int Mic"},
+	{"IN2P", NULL, "Int Mic"},
+	{"IN2N", NULL, "Int Mic"},
+	{"IN2P", NULL, "micbias1"},
 	{"Headphone", NULL, "HPOL"},
 	{"Headphone", NULL, "HPOR"},
 	{"Ext Spk", NULL, "SPOL"},
 	{"Ext Spk", NULL, "SPOR"},
+
 	{ "AIF1 Playback", NULL, "ssp2 Tx"},
 	{ "ssp2 Tx", NULL, "codec_out0"},
 	{ "ssp2 Tx", NULL, "codec_out1"},
-- 
1.9.1

